src/app/user/login/login.page.html

```diff
@@ -1,16 +1,50 @@
-<form>
+<form #loginForm="ngForm" (submit)="onLogin(loginForm)">
+
   <div class="auth-wrapper">
+
     <div class="auth-input">
-      <ion-input type="email" fill="solid" label="Имейл*" labelPlacement="floating" clearInput></ion-input>
+      <ion-input type="text" fill="solid" label="Имейл*" name="email" labelPlacement="floating" clearInput ngModel
+        #email="ngModel" required pattern="^[a-zA-Z0-9\._+\-]+@[a-zA-Z0-9\.]+\.[a-zA-Z]{2,}$">
+      </ion-input>
     </div>
 
+    <ng-container *ngIf="email.touched">
+      <p class="auth-validation-error-message" *ngIf="email.errors?.['required']">
+        Полето е задължително за попълване!
+      </p>
+      <p class="auth-validation-error-message" *ngIf="email.errors?.['pattern']">
+        Моля въведете валиден имейл!
+      </p>
+
+      <!-- TODO: When server returns email already taken -->
+      <!-- <p class="auth-validation-error-message" *ngIf="emailError">
+        Този имейл вече е използван!
+      </p> -->
+    </ng-container>
+
     <div class="auth-input">
-      <ion-input type="password" fill="solid" label="Парола*" labelPlacement="floating" clearInput></ion-input>
+      <ion-input type="password" fill="solid" label="Парола*" name="password" labelPlacement="floating" clearInput
+        ngModel #password="ngModel" required minlength="8"></ion-input>
     </div>
+
+    <ng-container *ngIf="password.touched">
+      <p class="auth-validation-error-message" *ngIf="password.errors?.['required']">
+        Полето е задължително за попълване!
+      </p>
+      <p class="auth-validation-error-message" *ngIf="password.errors?.['minlength']">
+        Парола трябва да е с дължина поне 8 символа!
+      </p>
+
+      <!-- TODO: When server returns email already taken -->
+      <!-- <p class="auth-validation-error-message" *ngIf="passwordError">
+        Грешна парола!
+      </p> -->
+    </ng-container>
+
   </div>
 
   <div class="main-button-wrapper">
-    <button class="main-button">Вход</button>
+    <button class="main-button" [disabled]="loginForm.invalid" #button>Вход</button>
```

Comment:
Why "#button" ? Using the element in the ts file or setting an ID ? Also check if it's like this, after clicking, if the page will reload? (It shouldn't)

Reply:
#button is not needed. The form is not reloading the page